### PR TITLE
completed task 11

### PR DESCRIPTION
**Pull request by nearhanfarhan**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -79,6 +79,49 @@ describe("far-han-news tests", () => {
...
+      });
+    });
+    test('GET 400: should return an error if a forbidden order is used', () => {
+      return request(app)
+      .get("/api/articles?order=bananas")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+    });
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:134]: What happens if there is a valid sort/existing topic with no articles? It should be a 200, like when you get an article with no comments. Not an error, but should be tested that it's happy on the 200.

File: models/articles.models.js [hunk 2]
```diff
@@ -11,19 +11,49 @@ exports.fetchArticleById = (article_id) => {
...
-  return db.query(text).then(({ rows }) => {
+  const acceptedOrders = ["asc", "desc"];
+
+  if (!acceptedSorts.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  if (!acceptedOrders.includes(order)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
```
Comment (dappernerddesigns) [inline models/articles.models.js:33]: Great to see the green listing here front and centre.

File: models/articles.models.js [hunk 3]
```diff
@@ -11,19 +11,49 @@ exports.fetchArticleById = (article_id) => {
...
+  text += `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id `;
+
+  if (topic) {
+    text += `WHERE topic = $1 `;
+    params.push(topic);
+  }
+
+  text += `GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`;
+  return db.query(text, params).then(({ rows }) => {
+    if (rows.length===0){
```
Comment (dappernerddesigns) [inline models/articles.models.js]: This is something to look at for the missing test I mentioned in your test suite.

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
Really good work on this, one test missing on this that might throw your promise reject for a loop. Have a look at the comments then re submit it when you're ready.

File: __tests__/app.test.js (hunk 1 above)
Comment (nearhanfarhan) [reply __tests__/app.test.js:134]: I see, just noticed the topics table

**Review by aloush-dev — APPROVED** [review]
Nice job on this one!
Merge away

**Pull request merged**